sound/soc/soc-pcm.c

```diff
@@ -1534,7 +1534,7 @@ int dpcm_be_dai_startup(struct snd_soc_pcm_runtime *fe, int stream)
 			be->dpcm[stream].state = SND_SOC_DPCM_STATE_CLOSE;
 			goto unwind;
 		}
-
+		be->dpcm[stream].be_start = 0;
```

Comment:
Should we protect the be_start here as well?

Reply:
I am not sure: I looked and there's already a mutex, see my earlier answer

I looked at the code and only saw a couple of cases where the spinlock is not used, e.g.

dpcm_be_dai_startup()
dpcm_be_dai_stop()
pcm_be_dai_hw_free()
dpcm_be_dai_hw_free()

but for all of them there is a mutex anyways in the caller, e.g.

static int dpcm_fe_dai_hw_params(struct snd_pcm_substream *substream,
				 struct snd_pcm_hw_params *params)
{
	struct snd_soc_pcm_runtime *fe = asoc_substream_to_rtd(substream);
	int ret, stream = substream->stream;

	mutex_lock_nested(&fe->card->mutex, SND_SOC_CARD_CLASS_RUNTIME);

So I would assume the risk is only between triggers where the mutex is not used.